File: sentry/src/test/java/io/sentry/servlet/SentryServletContainerInitializerTest.java
package io.sentry.servlet;

import io.sentry.BaseTest;
import org.junit.Test;

import javax.servlet.ServletContainerInitializer;
import javax.servlet.ServletContext;
import java.util.ServiceLoader;

import static org.hamcrest.MatcherAssert.assertThat;
import static org.hamcrest.Matchers.contains;
import static org.hamcrest.Matchers.instanceOf;
import static org.mockito.ArgumentMatchers.eq;
import static org.mockito.Mockito.mock;
import static org.mockito.Mockito.verify;

public class SentryServletContainerInitializerTest extends BaseTest {

    @Test
    public void testInitializerInjectedViaServiceLoader() throws Exception {
        ServiceLoader<ServletContainerInitializer> serviceLoader = ServiceLoader.load(ServletContainerInitializer.class);
        assertThat(serviceLoader, contains(instanceOf(SentryServletContainerInitializer.class)));
    }

    @Test
    public void testFilterAddedToServletContext() throws Exception {
        ServletContext mockServletContext = mock(ServletContext.class);
        new SentryServletContainerInitializer().onStartup(null, mockServletContext);

        verify(mockServletContext).addListener(eq(SentryServletRequestListener.class));
    }
}

File: sentry-servlet-jakarta/src/test/kotlin/io/sentry/servlet/jakarta/SentryServletContainerInitializerTest.kt
package io.sentry.servlet.jakarta

import jakarta.servlet.ServletContext
import org.mockito.kotlin.check
import org.mockito.kotlin.mock
import org.mockito.kotlin.verify
import java.util.EventListener
import kotlin.test.Test
import kotlin.test.assertEquals

class SentryServletContainerInitializerTest {
    private val initializer =
        SentryServletContainerInitializer()

    @Test
    fun `adds SentryServletRequestListener on startup`() {
        val servletContext = mock<ServletContext>()
        initializer.onStartup(null, servletContext)
        verify(servletContext).addListener(
            check { it: Class<out EventListener> ->
                assertEquals(SentryServletRequestListener::class.java, it)
            },
        )
    }
}
